src/chapters/exceptions/handling.rst

```diff
@@ -0,0 +1,189 @@
+not handled in compiling, it is passed to runtime and called an **unchecked exception**. C# does not contain checked exceptions 
+and therefore it is up to you, the programmer to decide what to do to handle one when the need arises in runtime.
+
+Here's some advice to consider when contemplating when to handle an exception. This comes from Karl Seguin's 
+`Foundations of Programming <https://www.openmymind.net/FoundationsOfProgramming.pdf>`__.
+
+#. Only handle exceptions that you can actually do something about.
+#. You can't do anything about the vast majority of exceptions.
+
+For example, if your code cannot connect to a database, there is probably nothing your program can 
+do about it. However, as we allude to on the previous page, if you receive invalid input from a user, you can still throw an 
+exception, re-prompt them to refine their input with an error message to help them get it right the next time.
+
+.. index:: ! try, ! catch, ! finally
+
+Try/Catch/Finally
+-----------------
+
+To handle exceptions in C#, we use the **try/catch/finally** construction. This functions very similarly 
+to exception handling tooling in other languages. If you expect that a method might return an exception, the method call should 
+be wrapped in a ``try`` statement. This could be because you threw the exception in the method yourself. Or you are in one of 
+the common exception scenarios we outlined on the previous page.
+
+If the code runs without throwing an exception, then the program continues as usual. However, if the ``try`` block results in an 
+exception, then a later ``catch`` statement will be called. ``catch`` gives the program instructions on what to do in the 
+event of an exception. The ``catch`` block prevents the program from stopping when it reaches the exception. 
+
+Here's how we can update our Temperature constructor with a ``try/catch`` to handle the exception:
+
+.. sourcecode:: c#
+   :linenos:
+
+   public Temperature(double fahrenheit)
+   {
+      try
+      {
+         Fahrenheit = fahrenheit;
+      }
+      catch(ArgumentOutOfRangeException e)
+      {
+         Fahrenheit = -459.67;
+      }
+   }
+
+The first action the constructor method does is call the Fahrenheit's setter method. If invoking the 
+setter does not result in an exception being thrown, then the new ``Temperature`` object is created and
+given an initial Fahrenheit value. If the action inside the ``try`` block results in an exception,
+or specifically an ``ArgumentOutOfRange`` instance, then the ``catch`` block runs and the initial 
+value of a new ``Temperature`` object's ``fahrenheit`` field is set to absolute zero. 
+
+Although in this example, we don't explicitly do anything with the argument ``e`` ("e" for exception),
+there are some scenarios where you may wish to display the exception or it's message in the view of your
+running app. It also an option to rethrow an exception after it has been caught. You won't need to rethrow 
```

Comment:
This sentence is really long. Perhaps break it up like so: "With this example, we don't explicitly do anything with the argument ``e`` ("e" for exception). However, there are scenarios ...."

Reply:
good note!